### PR TITLE
fix(table): replace unrecoverable panic with error

### PR DESCRIPTION
**Pull request by AveralS**
## Problem

I am using badger to store some temporary transactional data. My database crashed due to power failure and when the app starts again it panics due to the broken badger database. I can't handle this panic, because it is called in a separate goroutine.

## Solution

[initBiggestAndSmallest method](https://github.com/dgraph-io/badger/blob/ef0e55289cf70ade33f1c1dd19dcd8eced85cbbb/table/table.go#L346) should return an error when recovering from underlying panic.

### CODE REVIEW

**Review by mangalaman93 — DISMISSED** [review]

**Comment by mangalaman93** [comment]
@AveralS Thanks for the PR. Could you please sign the CLA?

**Comment by AveralS** [comment]
> @AveralS Thanks for the PR. Could you please sign the CLA?

No, i dont see accept button here

image

**Comment by mangalaman93** [comment]
That's pretty odd. You could try a different browser.

File: table/table.go [hunk 1]
```diff
@@ -343,14 +343,14 @@ func OpenInMemoryTable(data []byte, id uint64, opt *Options) (*Table, error) {
 	return t, nil
 }
 
-func (t *Table) initBiggestAndSmallest() error {
+func (t *Table) initBiggestAndSmallest() (err error) {
```
Comment (harshil-goel) [inline table/table.go:346]: Can you remove the named return variable?

**Pull request closed without merging**